compiler/modules/print.llvm/llvm.bal

```diff
@@ -314,20 +323,74 @@ public class FunctionDefn {
         else {
             self.variableNames[regName] = 1;
         }
+        if name is string && !self.nameValid(regName) {
+            regName = "\"" + regName + "\"";
+        }
         string reg = "%" + regName;
         return reg;
-    } 
+    }
+
+    // Verify name is a valid LLVM name otherwise must escape
+    function nameValid(string name) returns boolean{
```

Comment:
This whole function needs rewriting.

Reply:
Function replace with new `escapeName` function